lib/matplotlib/testing/compare.py

```diff
@@ -128,6 +132,60 @@ def convert(old, new):
     return convert
 
 
+class _SVGConverter(object):
+    def __init__(self):
+        self._proc = None
+        # We cannot rely on the GC to trigger `__del__` at exit because
+        # other modules (e.g. `subprocess`) may already have their globals
+        # set to `None`, which make `proc.communicate` or `proc.terminate`
+        # fail.  By relying on `atexit` we ensure the destructor runs before
+        # `None`-setting occurs.
+        atexit.register(self.__del__)
```

Comment:
Yes, but `__del__` can safely be executed twice (a file stream can be closed multiple times).